packages/store/src/deploymentsSlice/thunks.ts

```diff
@@ -115,96 +143,120 @@ const trackDeploymentCreation = (totalDeploymentCount, hasDeployments, trial_exp
 };
 
 const MAX_PREVIOUS_PHASES_COUNT = 5;
-export const createDeployment = createAsyncThunk(`${sliceName}/createDeployment`, ({ newDeployment, hasNewRetryDefault = false }, { dispatch, getState }) => {
-  let request;
-  if (newDeployment.filter_id) {
-    request = GeneralApi.post(`${deploymentsApiUrlV2}/deployments`, newDeployment);
-  } else if (newDeployment.group) {
-    request = GeneralApi.post(`${deploymentsApiUrl}/deployments/group/${newDeployment.group}`, newDeployment);
-  } else {
-    request = GeneralApi.post(`${deploymentsApiUrl}/deployments`, newDeployment);
+export const createDeployment = createAppAsyncThunk(
+  `${sliceName}/createDeployment`,
+  (
+    {
+      newDeployment,
+      hasNewRetryDefault = false
+    }: { hasNewRetryDefault: boolean; newDeployment: NewDeployment | (NewDeploymentForGroup & { group: string }) | NewDeploymentV2 },
+    { dispatch, getState }
+  ) => {
+    let request;
+    if ('filter_id' in newDeployment && newDeployment.filter_id) {
+      request = GeneralApi.post(`${deploymentsApiUrlV2}/deployments`, newDeployment);
+    } else if ('group' in newDeployment && newDeployment.group) {
+      request = GeneralApi.post(`${deploymentsApiUrl}/deployments/group/${newDeployment.group}`, newDeployment);
+    } else {
+      request = GeneralApi.post(`${deploymentsApiUrl}/deployments`, newDeployment);
+    }
+    const totalDeploymentCount = (Object.values(getDeploymentsByStatusSelector(getState())) as DeploymentStatus[]).reduce<number>(
+      (accu, item) => accu + item.total,
+      0
+    );
+    const { hasDeployments } = getGlobalSettings(getState());
+    const { trial_expiration } = getOrganization(getState());
+    return request
+      .catch(err => commonErrorHandler(err, 'Error creating deployment.', dispatch))
+      .then(data => {
+        const lastslashindex = data.headers.location.lastIndexOf('/');
+        const deploymentId = data.headers.location.substring(lastslashindex + 1);
+        const deployment = {
+          ...newDeployment,
+          id: deploymentId,
+          devices: 'devices' in newDeployment && newDeployment.devices ? newDeployment.devices.map(id => ({ id, status: 'pending' })) : [],
```

Comment:
Maybe
```suggestion
          devices: newDeployment.hasOwnProperty('devices') ? newDeployment.devices.map(id => ({ id, status: 'pending' })) : [],
```
to stay consistent with similar checks in the codebase 👌.

Reply:
unfortunately,  `hasOwnProperty` is not currently providing the type narrowing, so we would need to alter the interface of the function to something like this: 
```
interface Object {
  hasOwnProperty<K extends PropertyKey>(key: K): this is Record<K, unknown>;
}
```
  
'prop' in obj is a current suggested type guard for typescript  

I am ok with both approaches, but I would probably change it in a separate task (maybe the one for type improvement)

Reply:
😞 nah, no need to change this, but I would still consider the `in` usage risky/ a bad practice... and yes, that might be a good addition to the type improvement to ensure we don't rely on `in` if possible/ we have types used more broadly (and then maybe https://github.com/microsoft/TypeScript/issues/44253#issuecomment-2619832197 to not rely on it being a `Record` or `unknown`).